README.md

```diff
@@ -85,23 +78,55 @@ You may also need to do another query that depends on the first one to get the r
 
 ```ruby
 resolve -> (obj, args, context) do
-  FindQuery.new(Product, args["id"]) do |product|
-    FindQuery.new(Image, product.image_id)
+  RecordLoader.for(Product).load(args["id"]).then do |product|
+    RecordLoader.for(Image).load(product.image_id)
   end
 end
 ```
 
-If the second query doesn't depend on the other one, then you can use GraphQL::Batch::QueryGroup, which allows each query in the group to be batched with other queries.
+If the second query doesn't depend on the first one, then you can use GraphQL::Batch.all, which allows each query in the group to be batched with other queries.
 
 ```ruby
 resolve -> (obj, args, context) do
-  smart_collection_query = CountQuery.new(SmartCollection, context.shop_id)
-  custom_collection_query = CountQuery.new(CustomCollection, context.shop_id)
-
-  QueryGroup.new([smart_collection_query, custom_collection_query]) do
-    smart_collection_query.result + custom_collection_query.result
+  GraphQL::Batch.all([
+    CountLoader.for(Shop, :smart_collections).load(context.shop_id),
+    CountLoader.for(Shop, :custom_collections).load(context.shop_id),
+  ]).then do |results|
+    results.reduce(&:+)
   end
 end
+
+`.then` can optionally take two lambda arguments, the first of which is equivalent to passing a block to `.then`, and the second one handles exceptions.  This can be used to provide a fallback
```

Comment:
what happens if you give `then` a first argument *and* a block?

wording tweaks:
> `.then` can optionally take two lambda arguments instead of a block. The first argument takes the place of a block (being called on success) and the second is called when an exception occurs. This can be used to provide a fallback: